CombinatorialGames/Game/Loopy/Outcome.lean

```diff
@@ -0,0 +1,255 @@
+  classical rw [rightOutcome]
+  split_ifs with w l
+  on_goal 3 => simpa using ⟨w, l⟩
+  all_goals rw [false_iff]
+  exacts [(·.1 w), (·.2 l)]
+
+/-- If there is no infinite play starting by from `x` with right going second,
+then `x` cannot end in a draw with right going second. -/
+-- Note: the spelling `Relation.Comp (· ∈ ·.rightMoves) (· ∈ ·.leftMoves)` is slightly longer
+theorem not_isLeftDraw_of_acc_comp {x : LGame}
+    (h : Acc (fun z x ↦ ∃ y ∈ x.leftMoves, z ∈ y.rightMoves) x) : ¬ IsLeftDraw x :=
+  h.rec fun _x _ ih ⟨nw, nl⟩ ↦
+    have ⟨y, hyx, hy⟩ := not_isLeftLoss_iff_exists.mp nl
+    have ⟨z, hzy, hz⟩ := not_isRightLoss_iff_exists.mp (not_isLeftWin_iff_forall.mp nw y hyx)
+    ih z ⟨y, hyx, hzy⟩ ⟨hz, not_isRightWin_iff_forall.mp hy z hzy⟩
+
+/-- If there is no infinite play starting from `x` with right going first,
+then `x` cannot end in a draw with right going first. -/
+theorem not_isRightDraw_of_acc_comp {x : LGame}
+    (h : Acc (fun z x ↦ ∃ y ∈ x.rightMoves, z ∈ y.leftMoves) x) : ¬ IsRightDraw x :=
+  h.rec fun _x _ ih ⟨nw, nl⟩ ↦
+    have ⟨y, hyx, hy⟩ := not_isRightLoss_iff_exists.mp nl
+    have ⟨z, hzy, hz⟩ := not_isLeftLoss_iff_exists.mp (not_isRightWin_iff_forall.mp nw y hyx)
+    ih z ⟨y, hyx, hzy⟩ ⟨hz, not_isLeftWin_iff_forall.mp hy z hzy⟩
+
+theorem not_isLeftDraw_of_acc_isOption {x : LGame} (h : Acc IsOption x) : ¬ IsLeftDraw x :=
+  not_isLeftDraw_of_acc_comp <| Subrelation.accessible
+    (fun ⟨_, hyx, hzy⟩ ↦ .tail (.single <| .inr hzy) (.inl hyx)) h.transGen
+
+theorem not_isRightDraw_of_acc_isOption {x : LGame} (h : Acc IsOption x) : ¬ IsRightDraw x :=
+  not_isRightDraw_of_acc_comp <| Subrelation.accessible
+    (fun ⟨_, hyx, hzy⟩ ↦ .tail (.single <| .inl hzy) (.inr hyx)) h.transGen
+
+end LGame
+
+namespace IGame
+
+open LGame
+
+theorem not_isLeftDraw (x : IGame) : ¬ IsLeftDraw x :=
+  LGame.not_isLeftDraw_of_acc_isOption x.acc_toLGame
+
+theorem not_isRightDraw (x : IGame) : ¬ IsRightDraw x :=
+  LGame.not_isRightDraw_of_acc_isOption x.acc_toLGame
+
+private theorem win_loss_iff {x : IGame} :
+    ((IsLeftWin x ↔ 0 ⧏ x) ∧ (IsLeftLoss x ↔ x ≤ 0)) ∧
+    ((IsRightWin x ↔ x ⧏ 0) ∧ (IsRightLoss x ↔ 0 ≤ x)) :=
+  x.moveRecOn fun x hl hr ↦ by
+    rw [isLeftWin_iff_exists, isLeftLoss_iff_forall, isRightWin_iff_exists, isRightLoss_iff_forall,
+      zero_lf, lf_zero, le_zero, zero_le, leftMoves_toLGame, rightMoves_toLGame]
+    simp_rw [Set.forall_mem_image, Set.exists_mem_image]
+    constructor <;>
+      refine ⟨exists_congr fun y ↦ and_congr_right fun hy ↦ ?_, forall₂_congr fun y hy ↦ ?_⟩
+    exacts [(hl y hy).2.2, (hl y hy).2.1, (hr y hy).1.2, (hr y hy).1.1]
+
+theorem isLeftWin_iff_zero_lf {x : IGame} : IsLeftWin x ↔ 0 ⧏ x := win_loss_iff.1.1
+theorem isLeftLoss_iff_le_zero {x : IGame} : IsLeftLoss x ↔ x ≤ 0 := win_loss_iff.1.2
+theorem isRightWin_iff_lf_zero {x : IGame} : IsRightWin x ↔ x ⧏ 0 := win_loss_iff.2.1
+theorem isRightLoss_iff_zero_le {x : IGame} : IsRightLoss x ↔ 0 ≤ x := win_loss_iff.2.2
```

Comment:
I think by mathlib convention these should be
```suggestion
theorem isLeftLoss_iff_nonpos {x : IGame} : IsLeftLoss x ↔ x ≤ 0 := win_loss_iff.1.2
theorem isRightWin_iff_lf_zero {x : IGame} : IsRightWin x ↔ x ⧏ 0 := win_loss_iff.2.1
theorem isRightLoss_iff_nonneg {x : IGame} : IsRightLoss x ↔ 0 ≤ x := win_loss_iff.2.2
```
but I'm not sure what convention this repo is following. It has both [IGame.sub_nonneg](https://vihdzp.github.io/combinatorial-games/CombinatorialGames/Game/Basic.html#IGame.sub_nonneg) and [IGame.zero_le_intCast](https://vihdzp.github.io/combinatorial-games/CombinatorialGames/Game/Basic.html#IGame.zero_le_intCast), [IGame.zero_le](https://vihdzp.github.io/combinatorial-games/CombinatorialGames/Game/IGame.html#IGame.zero_le), `IGame.zero_le_neg` etc.

Reply:
We generally prefer `zero_le` / `zero_lt` / etc. on `IGame`. It's misleading to use `nonneg` / `pos` / etc. outside of a linear order; for instance, star is neither positive > 0 nor nonpositive ≤ 0.

`IGame.sub_nonneg` was just copying the Mathlib name `sub_nonneg`. I wouldn't mind a PR to rename it.